misc/userscripts/password_fill

```diff
@@ -209,6 +210,78 @@ choose_entry_zenity_radio() {
     choose_entry_menu || true
 }
 
+# Default backend selection.
+backend="pass"
+
+# =======================================================
+# backend: LastPass. 
+# The LastPass backend depnds only on a system installation of lastpass-cli.
+# Users must define their lastpass username in password_fill_rc and set 
+# the backend variable to "last_pass"
+# e.g.
+#
+# #!/usr/bin/env bash
+# username=lp-username@email.com
+# backend=last_pass
+#
+# Temporary files are created that store account user name along
+# with the lastpass id for a given address. Users can redefine where the
+# temporary files will be saved with LP_CACHE_PREFIX the default is as such.
+#
+# No password data is ever written to a file with this method.
+
+LP_CACHE_PREFIX=$HOME/.local/share/qutebrowser/userscripts/id_cache
+
+last_pass_backend() {
+    init() {
+      PREFIX=$LP_CACHE_PREFIX
+      set +e
+
+      # Remove any old LP temporary id files.
+      for f in $PREFIX/; do
+        [ -f $f ] && rm $f 
+      done
+
+      # Check to see if the system is logged into LP else prompt
+      # the user with the LP login.
+      lpass ls
+      if ! [ $? -eq 0 ]; then
+        lpass login --trust $username
+        lpass ls
+        if [ $? -eq 0 ]; then
+          msg "Lastpass successful login."
+        else
+          die "Lastpass failed to initialize correctly."
+        fi
+      fi
+      set -e
+    }
+
+    query_entries() {
+        local url="$1"
+
+        set +e
+        lpass show $url
+        if [ $? -eq 0 ]; then
+            # Save files as usernames w/ corresponding LP id token.
+            for id in $(lpass show --id -x $url); do
+                echo "$id" > $PREFIX/$(lpass show --username -x $id)
```

Comment:
If `$PREFIX` path is not an existing directory, we blow up

Reply:
I've moved the default prefix path to `.config/qutebrowser/id_cache` and added functionality to try to make the directory if it DNE. It seems reasonable to expect at least `.config/qutebrowser` will exist as password_fill relies on `password_fill_rc` to exist at that location.